pkg/server/updateResults.go

```diff
@@ -45,43 +48,87 @@ func updateResultRequest(ce *CreateExperiment) UpdateResults {
-					{
-						Type:      "deployment",
-						Name:      kubeobj.Name,
-						Namespace: kubeobj.Namespace,
-						Containers: []model.ContainerMetrics{
-							{
-								ContainerImage: contlist.ContainerImage,
-								ContainerName:  contlist.ContainerName,
-								Metrics:        metricsList,
+
+			endTimes := getTimeWindows(1, int(windowSizeInt))
+			for i := 1; i < len(endTimes); i++ {
+				startTime := endTimes[i-1]
+				endTime := endTimes[i]
+				unixTime := endTime.Unix()
+				// get queries from performanceProfile per container:
+				metricsList := pm.GetPerformanceProfileInstanceMetrics(ce.ClusterName, kubeobj.Namespace,
+					kubeobj.Name, contlist.ContainerName, ce.TrialSettings.MeasurementDuration, unixTime)
+
+				updateResult = UpdateResults{
+					Version:        ce.Version,
+					ExperimentName: ce.ExperimentName,
+					StartTimestamp: startTime.Format("2006-01-02T15:04:05.000Z"), //an hour ago from now
+					EndTimestamp:   endTime.Format("2006-01-02T15:04:05.000Z"),   //now
+					KubernetesObjects: []model.KubernetesObjectMetrics{
+						{
+							Type:      "deployment",
+							Name:      kubeobj.Name,
+							Namespace: kubeobj.Namespace,
+							Containers: []model.ContainerMetrics{
+								{
+									ContainerImage: contlist.ContainerImage,
+									ContainerName:  contlist.ContainerName,
+									Metrics:        metricsList,
+								},
 							},
 						},
 					},
-				},
+				}
+				updateResults = []UpdateResults{updateResult}
+				upJson, _ := json.Marshal(updateResults)
+				klog.V(9).Infof("Created updateResults Object %v", string(upJson))
+				err := postUpdateResult(updateResults)
+				count := 0
+				for err != nil && count < config.Cfg.RetryCount {
+					count = count + 1
+					klog.Errorf("Cannot post updateResult %s in kruize: Will retry \n", ce.ExperimentName)
+					time.Sleep(time.Duration(config.Cfg.RetryInterval) * time.Millisecond)
+					err = postUpdateResult(updateResults)
+				}
+
 			}
-			updateResults = append(updateResults, updateResult)
 
 		}
 
 	}
-	up, _ := json.Marshal(updateResults)
-	klog.Infof("Processed %s", ce.ExperimentName)
-	klog.V(9).Infof("Created updateResults Object %v", string(up))
-	return updateResult
+}
+func postUpdateResult(updateResults []UpdateResults) error {
+	//post updateResults request to kruize
+	requestBodyJSON, err := json.Marshal(updateResults)
+	if err != nil {
+		klog.Error("Error encoding JSON:", err)
+		return err
+	}
+	client := utils.HTTPClient()
+	klog.V(9).Infof("Posting updateResult to Kruize Service %v", bytes.NewBuffer(requestBodyJSON))
+	res, err := client.Post(update_results_url, "application/json", bytes.NewBuffer(requestBodyJSON))
+	if err != nil {
+		return err
+	} else if res.StatusCode == 201 {
+		klog.V(5).Infof("Successful updateResults reqest for request %s , startTime: %s , endTime:%s", updateResults[0].ExperimentName, updateResults[0].StartTimestamp, updateResults[0].EndTimestamp)
+		return nil
+	}
+	return nil
```

Comment:
- redundant return inside else.
- Is it possible to get any other return status? I'd log it just in case.
```suggestion
	} else if res.StatusCode == 201 {
		klog.V(5).Infof("Successful updateResults reqest for request %s , startTime: %s , endTime:%s", updateResults[0].ExperimentName, updateResults[0].StartTimestamp, updateResults[0].EndTimestamp)
	} else {
		klog.Warnf("Received unexpected status code( s) from update request.", res.StatusCode)
	}
	return nil
```